### PR TITLE
WATF Number parsing

### PR DESCRIPTION
**Pull request by maurobringolf**
This does not cover everything by far, but I think its a fair starting point where we can make separate issues for the remaining cases.

### CODE REVIEW

File: src/compiler/parsing/watf/number-literals.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+export function parse64F( sourceString: string ): number {
+  return parseFloat(sourceString);
+}
+
+export function parse32I( sourceString: string ): number {
+  let value = 0;
+  if (isHexLiteral(sourceString)) {
+    value = ~~parseInt(sourceString, 16);
+  } else if (isDecimalExponentLiteral(sourceString)) {
+
```
Comment (xtuc) [inline src/compiler/parsing/watf/number-literals.js]: Is this empty? Could we throw an error here, like: unsupported value...

File: test/compiler/parsing/fixtures/watf/number-literals/f32/decimal-negative-integer/actual.wast [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(module
+  (func (
+    (f32.const 33)
```
Comment (xtuc) [inline test/compiler/parsing/fixtures/watf/number-literals/f32/decimal-negative-integer/actual.wast]: This should probably be a negative number? According to the name of the test.

File: test/compiler/parsing/fixtures/watf/number-literals/f32/decimal-positive-float/actual.wast [hunk 3]
```diff
@@ -0,0 +1,5 @@
+(module
+  (func (
+    (f32.const -98989898)
```
Comment (xtuc) [inline test/compiler/parsing/fixtures/watf/number-literals/f32/decimal-positive-float/actual.wast]: And this should probably be position according the its name.

File: test/compiler/parsing/fixtures/watf/number-literals/f32/decimal-positive-integer/actual.wast [hunk 4]
```diff
@@ -0,0 +1,5 @@
+(module
+  (func (
+    (f32.const 0.3)
```
Comment (xtuc) [inline test/compiler/parsing/fixtures/watf/number-literals/f32/decimal-positive-integer/actual.wast]: It seems that float and int tests should be swapped.

File: test/compiler/parsing/fixtures/watf/number-literals/i32/hexadecimal-full/expected.json [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+          "result": null,
+          "body": [
+            {
+              "type": "Instr",
+              "id": "const",
+              "object": "i32",
+              "args": [
+                {
+                  "type": "NumberLiteral",
+                  "value": -1
```
Comment (xtuc) [inline test/compiler/parsing/fixtures/watf/number-literals/i32/hexadecimal-full/expected.json]: `0xFFFFFFFF` should fit in an i32?

File: src/compiler/parsing/watf/number-literals.js (hunk 1 above)
Comment (maurobringolf) [reply src/compiler/parsing/watf/number-literals.js]: This is a TODO for literals of the form `234e-3` with decimal exponents. I'll throw an error here and will create issues for all cases that Im aware of that are missing 👍

File: test/compiler/parsing/fixtures/watf/number-literals/i32/hexadecimal-full/expected.json (hunk 5 above)
Comment (xtuc) [reply test/compiler/parsing/fixtures/watf/number-literals/i32/hexadecimal-full/expected.json]: `0xFFFFFFFF | 0 === -1` lgtm then

**Pull request merged**